packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -1,18 +1,38 @@
 import { TransactionBaseService } from "./transaction-base-service"
 import {
+  Address,
   Cart,
   Customer,
   Payment,
   PaymentSession,
   PaymentSessionStatus,
+  ShippingMethod,
 } from "../models"
 import { PaymentService } from "medusa-interfaces"
-import { PaymentProviderDataInput } from "../types/payment-collection"
 
 export type Data = Record<string, unknown>
 export type PaymentData = Data
 export type PaymentSessionData = Data
 
+export type PaymentContext = {
+  cart: {
+    context: Record<string, unknown>
+    id: string
+    email: string
+    shipping_address: Address | null
+    shipping_methods: ShippingMethod[]
+  }
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+}
+
+export type PaymentSessionResponse = {
+  update_requests: { customer: Record<string, unknown> }
```

Comment:
suggestion (non-blocking): should we call this `customer_metadata` just to make it super clear what it is that the plugin can update?

Reply:
Oli asked me to move it. I thought it was to include it here 😅

Reply:
In fact, I might just have been a bit too fast. It was a thought from oli and i "thought" it was a request 😂

Reply:
@srindom i ve miss placed the comments. It was about the isDefined 🤓